src/Service/QueueManager.php

```diff
@@ -218,9 +218,30 @@ private function jobExists(string $jobUuid): Promise
     private function processBatch(): \Generator
     {
         $this->logger->debug('Processing batch');
-        yield $this->elasticSearch->bulkIndexJobs($this->batch, $this->flowConfig->getTube());
+        $result = yield $this->elasticSearch->bulkIndexJobs($this->batch, $this->flowConfig->getTube());
+        $successfullyIndexedJobs = $this->batch;
+        $notIndexedJobs = [];
+
+        if ($result['errors'] === true) {
+            $successfullyIndexedJobs = array_filter($this->batch, function ($job) use ($result) {
+                return $result['items'][$job->getUuid()]['index']['status'] === 201;
+            });
+
+            $notIndexedJobs = array_filter($this->batch, function ($job) use ($result) {
+                return $result['items'][$job->getUuid()]['index']['status'] !== 201;
+            });
+        }
+
+        foreach ($notIndexedJobs as $singleJob) {
+            $this->logger->error(
+                sprintf(
+                    'Job with UUID "%s" could not be indexed in ElasticSearch',
+                    $singleJob->getUuid()
+                )
+            );
+        }
 
-        foreach ($this->batch as $singleJob) {
+        foreach ($successfullyIndexedJobs as $singleJob) {
             yield $this->beanstalkClient->put(
```

Comment:
Dopo questa put bisogna mettere la loggata "successfully produced a job" che adesso sta in ProducerInstance se non ricordo male.

Reply:
I think we should leave the logging "successfully produced a job" in the ProducerInstance: if bulk index fails the QueueManager will log the error and stop, and the ProducerInstance will not log "successfully produced a job"

Reply:
Mmmm, are you sure? I think that the ProducerInstance, most of the times, will log "Successfully produced a job" even if the job has not been persisted on ES yet. This is because the ProducerInstance logs "successfully produced a job" immediately when the job is added to the batch if the max batch size is not reached and in such case the job is not persisted on ES yet. Am I wrong?